gobblin-core/src/main/java/gobblin/converter/filter/AvroSchemaFieldRemover.java

```diff
@@ -99,6 +99,9 @@ private Schema removeFields(Schema schema, Map<String, Schema> schemaMap) {
   }
 
   private Schema removeFieldsFromRecords(Schema schema, Map<String, Schema> schemaMap) {
+    Schema newRecord = Schema.createRecord(schema.getName(), schema.getDoc(), schema.getNamespace(), schema.isError());
```

Comment:
nit: This may be safe now because the processing is single-threaded but I think, in general, it is dangerous to put the new record schema in the map before fully constructed. Previously, it was constructed at the end. Why did you have to move it at the beginning?